lib/active_merchant/billing/gateways/priority.rb

```diff
@@ -0,0 +1,347 @@
+        params['merchantId'] = @options[:merchant_id]
+        params['paymentToken'] = authorization
+        params['shouldGetCreditCardLevel'] = true
+        params['source'] = options['source']
+        params['tenderType'] = options[:tender_type]
+
+        commit('capture', params: params, jwt: options)
+      end
+
+      def void(iid, options)
+        commit('void', iid: iid, jwt: options)
+      end
+
+      def verify(credit_card, jwt)
+        commit('verify', card_number: credit_card, jwt: jwt)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def get_payment_status(batch_id, options)
+        commit('get_payment_status', params: batch_id, jwt: options)
+      end
+
+      def close_batch(batch_id, options)
+        commit('close_batch', params: batch_id, jwt: options)
+      end
+
+      def create_jwt(options)
+        commit('create_jwt', params: @options[:merchant_id], jwt: options)
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]').
+          gsub(%r((number\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r((cvv\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]')
+      end
+
+      def add_bank(params, auth_code)
+        params['authCode'] = auth_code
+        params['authMessage'] = ''
+        params['authOnly'] = false
+        params['availableAuthAmount'] = 0
+        params['batch'] = nil
+        params['batchId'] = nil
+      end
+
+      def add_credit_card(params, credit_card, action, options)
+        return unless credit_card
+
+        card_details = {}
+
+        case action
+        when 'purchase'
+          card_details['avsStreet'] = options[:billing_address][:address1]
+          card_details['avsZip'] =  options[:billing_address][:zip]
+          card_details['cvv'] = credit_card.verification_value
+          card_details['entryMode'] = 'Keyed'
+          card_details['expiryDate'] = exp_date(credit_card)
+          card_details['expiryMonth'] = format(credit_card.month, :two_digits).to_s
+          card_details['expiryYear'] = format(credit_card.year, :two_digits).to_s
+          card_details['last4'] = nil
+          card_details['magstripe'] = nil
+          card_details['number'] = credit_card.number
+        when 'refund'
+          card_details['cardId'] = credit_card['cardId']
+          card_details['cardPresent'] = credit_card['cardPresent']
+          card_details['cardType'] = credit_card['cardType']
+          card_details['entryMode'] = credit_card['entryMode']
+          card_details['expiryMonth'] = credit_card['expiryMonth']
+          card_details['expiryYear'] = credit_card['expiryYear']
+          card_details['hasContract'] = credit_card['hasContract']
+          card_details['isCorp'] = credit_card['isCorp']
+          card_details['isDebit'] = credit_card['isDebit']
+          card_details['last4'] = credit_card['last4']
+          card_details['token'] = credit_card['token']
```

Comment:
Looks like `expiry*`, `entryMode`, `last4` are common to each action, so we can probably handle them in 1 place. Also, the `refund` portion is pulling values from `credit_card` as though it's a hash whereas in `purchase` it is pulling them as though they are an attribute of a payment object. I think the way they are handled in the `purchase` action is the correct way, but we should conform them to one or the other.